### PR TITLE
WIP: Improvements

### PR DESCRIPTION
**Pull request by zpawn**
Story https://app.shortcut.com/deskpro/story/161047/wrike-improvements

### CODE REVIEW

File: src/components/Link/Task.tsx [hunk 1]
```diff
@@ -162,9 +166,12 @@ export const LinkTask = () => {
...
                       metadata={TaskJson.link}
                       idKey={TaskJson.idKey}
-                      internalChildUrl={`/view/task/`}
                       externalChildUrl={TaskJson.externalUrl}
-                      childTitleAccessor={(e) => e.title}
+                      childTitleAccessor={(e) => (
+                        <ButtonAsLink type="button" onClick={() => onSelectTask(task as unknown as ITask)}>
+                          {e.title}
+                        </ButtonAsLink>
+                      ) as unknown as string}
```
Comment (HappyPaul55) [inline src/components/Link/Task.tsx:174]: We should avoid using `as` where possible to ensure type accuracy. It looks like we're typing a `ReactNode` as a `string` which seems questionable. Also why do we need to type a task to `ITask` via `unknown` when `task` should already be represented as an `ITask` (assuming that's what it is)

**Review by HappyPaul55 — CHANGES_REQUESTED** [review]
Apart from the type juggling, this looks good.

File: src/components/Link/Task.tsx (hunk 1 above)
Comment (zpawn) [reply src/components/Link/Task.tsx:174]: When I'll refactor this page, this terrible FieldMapping component will be removed, and this type juggling will no longer be necessary.

**Review by HappyPaul55 — APPROVED** [review]
Approved on the basis the types will be improved in a near future PR.

**Comment by AshleyDawson** [comment]
Closing due to incorrect build license

**Pull request closed without merging**